.github/workflows/ci.yml

```diff
@@ -7,79 +7,44 @@ on:
 name: Continuous integration
 
 jobs:
-  ubuntu-build-x86_64:
-    name: Ubuntu x86_64
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-        with:
-          submodules: recursive
-      - name: Cache CMake build directory
-        uses: actions/cache@v4
-        with:
-          path: |
-            build
-          key: ${{ runner.os }}-${{ runner.arch }}-cmake-build-${{ github.sha }}
-          restore-keys: |
-            ${{ runner.os }}-${{ runner.arch }}-cmake-build-
-      - name: Install dependencies
-        run: sudo apt-get update && sudo apt-get install -y
-          libasound2-dev
-          libgl-dev
-          libglu1-mesa-dev
-          libgtk-3-dev
-          libjack-dev
-          libmad0-dev
-          libpulse-dev
-          libudev-dev
-          libxinerama-dev
-          libx11-dev
-          libxrandr-dev
-          libxtst-dev
-          nasm
-      - name: Configure
-        run: cmake -B build -DWITH_FFMPEG_JOBS="$(nproc)"
-      - name: Build
-        run: cmake --build build --parallel "$(nproc)"
-
-  ubuntu-build-arm:
-    name: Ubuntu ARM
-    runs-on: ubuntu-24.04-arm
-    steps:
-      - uses: actions/checkout@v4
-        with:
-          submodules: recursive
-      - name: Cache CMake build directory
-        uses: actions/cache@v4
-        with:
-          path: |
-            build
-          key: ${{ runner.os }}-${{ runner.arch }}-cmake-build-${{ github.sha }}
-          restore-keys: |
-            ${{ runner.os }}-${{ runner.arch }}-cmake-build-
-      - name: Install dependencies
-        run: sudo apt-get update && sudo apt-get install -y
-          libasound2-dev
-          libgl-dev
-          libglu1-mesa-dev
-          libgtk-3-dev
-          libjack-dev
-          libmad0-dev
-          libpulse-dev
-          libudev-dev
-          libxinerama-dev
-          libx11-dev
-          libxrandr-dev
-          libxtst-dev
-          nasm
-      - name: Configure
-        run: cmake -B build -DWITH_FFMPEG_JOBS="$(nproc)" -DWITH_MINIMAID=OFF
-      - name: Build
-        run: cmake --build build --parallel "$(nproc)"
-
-  macos-build-arm64:
-    name: macOS (M1)
-    runs-on: macos-latest
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, ubuntu-24.04-arm, macos-latest, macos-13, windows-latest]
+        include:
+          - os: ubuntu-latest
```

Comment:
will this allow further expansion using docker containers in the future via [the `.container` setting](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idcontainer)? I'd like to include an older ubuntu target as suggested by @teejusb and fedora/arch/debian/suse to cover the other popular distributions in the future as well

Reply:
It sucks that the external actions execute in the context of the container, this make them less reusable. I like GitLab's approach better, where the setup steps just share the volume that's later attached to the requested container.

I think we can prepare a generic script to execute within the manually spawned container on the VM instead of using `container` setting in the future. How about not mixing the refactor with new additions? Let's do that in a separate PR later.